packages/vite/src/node/optimizer/index.ts

```diff
@@ -496,7 +496,9 @@ export async function runOptimizeDeps(
       // Write metadata file, delete `deps` folder and rename the `processing` folder to `deps`
       // Processing is done, we can now replace the depsCacheDir with processingCacheDir
       // Rewire the file paths from the temporal processing dir to the final deps cache dir
-      await removeDir(depsCacheDir)
+      if (fs.existsSync(depsCacheDir)) {
+        await removeDir(depsCacheDir)
+      }
```

Comment:
If the bug is within `fs.rmSync`, I think we should put the file existence check at:

https://github.com/vitejs/vite/blob/f8d9189fb99258a632a5345dafe789596620f944/packages/vite/src/node/utils.ts#L608-L610

instead. It would also be great to add a short comment as to why it's needed.

Reply:
ok